viz_scripts/generic_metrics.ipynb

```diff
@@ -101,192 +102,358 @@
+    "            bar_count += 1\n",
+    "            all_data_frames.append(df_sensed_tc)\n",
+    "    else:\n",
+    "        print(\"expanded_ct_sensed is empty\")\n",
+    "\n",
+    "    result_df = merge_dataframes(all_data_frames)\n",
+    "    stacked_bar_chart_generic(plot_title, result_df, file_name, bar_count)\n",
+    "    alt_text, alt_html = store_alt_text_stacked_bar_chart(result_df[result_df['Count'] > 0], file_name, plot_title)  \n",
     "except:\n",
-    "    generate_missing_plot(plot_title_no_quality,debug_df,file_name)\n",
-    "    alt_text = store_alt_text_missing(debug_df, file_name, plot_title_no_quality)"
+    "    generate_missing_plot(plot_title_no_quality,merged_debug_df,file_name)\n",
+    "    alt_text = store_alt_text_missing(merged_debug_df, file_name, plot_title_no_quality)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8fc63a45",
+   "metadata": {},
+   "source": [
+    "### 2. 100% Stacked Bar Charts (Represents 80th of the Number of Trips)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "about-seafood",
-   "metadata": {
-    "scrolled": false
-   },
+   "id": "77ece8ee",
+   "metadata": {},
    "outputs": [],
    "source": [
-    "plot_title_no_quality= \"Number of commute trips for each mode (selected by users)\"\n",
-    "file_name= 'ntrips_commute_mode_confirm%s' % file_suffix\n",
+    "def process_data_for_cutoff(data_frame, distance_col, mode_col, quality_func, label_title):\n",
+    "\n",
+    "    cutoff = data_frame.distance.quantile(0.8)\n",
+    "    if pd.isna(cutoff):\n",
+    "        cutoff = 0\n",
+    "\n",
+    "    dist_threshold = data_frame[distance_col].quantile(0.8).round(1)\n",
+    "    dist_threshold = str(dist_threshold)\n",
+    "\n",
+    "    labels = data_frame.loc[(data_frame['distance'] <= cutoff)][mode_col].value_counts(dropna=True).keys().tolist()\n",
+    "    values = data_frame.loc[(data_frame['distance'] <= cutoff)][mode_col].value_counts(dropna=True).tolist()\n",
+    "    processed_data = process_trip_data(labels, values, label_title)    \n",
+    "\n",
+    "    return processed_data,cutoff, dist_threshold\n",
+    "\n",
+    "plot_title_no_quality= \"Number of trips for 80th percentile for each mode (selected by users):\"\n",
+    "file_name = f'ntrips_under80{file_suffix}'\n",
     "\n",
     "try:\n",
-    "    if (len(dynamic_labels)):\n",
-    "        purpose_map_label =  scaffolding.mapping_labels(dynamic_labels, \"PURPOSE\")\n",
-    "        translation_work = purpose_map_label['work']\n",
-    "        trip_purpose_query = f\"Trip_purpose == '{translation_work}'\"\n",
+    "    bar_count = 0\n",
+    "    all_data_frames_u80 = []\n",
+    "    plot_title = plot_title_no_quality\n",
+    "    df_confirmed_tc_u80,cutoff, dist_threshold  = process_data_for_cutoff(expanded_ct, distance_col, 'Mode_confirm', scaffolding.get_quality_text, \"Labeled by user \\n (Based on Confirmed Trips)\")\n",
```

Comment:
I think we could just have "Labeled by user" and may not need "Confirmed Trips" - since they refer to the same thing, I think we could just choose one term and use it everywhere, what do you think?

Reply:
I'm thinking this might be a "less is more" situation since we are putting a lot of information on one chart. What would you think of text above the chart being something like
```
Number of trips for each mode 
Based on 667 user-labeled trips from 20 testers and participants
Of 5347 total trips from 27 users
```

Then the bar labels could be `Labeled Trips` and `Total Trips`

I know it's a little complicated to change the text, but I think it's probably worth it in terms of being as clear as possible with as little text as possible, later when "inferred" modes are introduced that would put us even closer to the text we were talking about in [the issue](https://github.com/e-mission/em-public-dashboard/issues/86#issuecomment-1960514964)

Do the text I suggested above seem clear to you? I know the difference between labeled and total trips is really important, so clarity comes first, but if at all possible I think shortening labels will be helpful.

Reply:
I agree on the approach of incorporating "less is more" in visualization, but it should not miss the information which is essential. I have a bit of concern regarding a scenarios with the above way -

1. When we provide the below text

```
Number of trips for each mode 
Based on 667 user-labeled trips from 20 testers and participants
Of 5347 total trips from 27 users
```
 and just show the bar labels as `Labeled Trips` and `Total Trips`, how would this showcase the bar representing `Total Trips` is actually showing the visualization for `Sensed by OpenPath`?

We would likely lose the representation stating that the bar representing `Total Trips` is in fact representing for `Sensed by OpenPath`.

Reply:
Note that this will be fixed when we introduce the inferred mode changes